### PR TITLE
lib/release: Fix version label logic for release cuts

### PR DESCRIPTION
**Pull request by justaugustus**
#### What type of PR is this?

/kind bug cleanup

#### What this PR does / why we need it:

- Move branch creation to RC cuts
- Enable beta cuts on master
- Disallow alpha releases after a beta has been cut

Continuation of https://github.com/kubernetes/release/pull/1286.
Tracking: https://github.com/kubernetes/release/issues/1207, https://github.com/kubernetes/release/issues/1281

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

/priority critical-urgent
/assign @saschagrunert 
cc: @kubernetes/release-engineering @kubernetes/release-managers 

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/release/issues/1207

#### Special notes for your reviewer:

~I'll cleanup on the commit(s) and PR title, description later; just getting this up to test against.~

#### Does this PR introduce a user-facing change?

```release-note
lib/release: Fix version label logic for release cuts
- Move branch creation to RC cuts
- Enable beta cuts on master
- Disallow alpha releases after a beta has been cut
```

### CODE REVIEW

**Comment by justaugustus** [comment]
Testing the following scenarios:


```shell
$ TOOL_ORG=justaugustus TOOL_BRANCH=rc-branch-cut krel gcbmgr --stage 
INFO Verifying repository state                   
INFO Repository is in clean state                 
INFO Found matching branch "rc-branch-cut"        
INFO Found matching organization "justaugustus" and repository "release" in remote: a (git@github.com:justaugustus/release.git) 
INFO Verifying remote HEAD commit                 
INFO Got remote commit: 1b3ef841bbb0ead1a10bdd53833362a8d2c0f807 
INFO Verifying that remote commit is equal to the local one 
INFO Repository is up-to-date                     
INFO Running gcbmgr with the following options: &{Stage:true Release:false Stream:false Branch:master ReleaseType:alpha BuildVersion: GcpUser: LastJobs:5 Repo:0xc000403930 Version:0xc000403940} 
INFO Build options: {  cloudbuild.yaml   kubernetes-release-test false false false   } 
INFO Retrieving Kubernetes release version for ci/latest on branch master 
INFO Retrieving Kubernetes build version from https://dl.k8s.io/ci/latest.txt... 
INFO Retrieved Kubernetes version: v1.19.0-alpha.3.261+96e13de777a9eb 
INFO Trying to get the kube-cross version for master... 
INFO Found the following kube-cross version: v1.13.9-5 
INFO Listing GCB substitutions prior to build submission... 
INFO BUILDVERSION: --buildversion=v1.19.0-alpha.3.261+96e13de777a9eb 
INFO TOOL_ORG: justaugustus                       
INFO TOOL_REPO: release                           
INFO TOOL_BRANCH: rc-branch-cut                   
INFO GCP_USER_TAG: stephen-k8s-at-agst-us         
INFO TYPE: alpha                                  
INFO RELEASE_BRANCH: master                       
INFO BUILD_AT_HEAD:                               
INFO NOMOCK_TAG:                                  
INFO NOMOCK:                                      
INFO TYPE_TAG: alpha                              
INFO BUILD_POINT: v1.19.0-alpha.3.261-96e13de777a9eb 
INFO KUBE_CROSS_VERSION: v1.13.9-5                
INFO Build directory: /home/augustus/go/src/k8s.io/release/gcb/stage 
INFO Config directory: /home/augustus/go/src/k8s.io/release/gcb/stage 
INFO cd-ing to build directory: /home/augustus/go/src/k8s.io/release/gcb/stage 
ID                                    CREATE_TIME                DURATION  SOURCE  IMAGES  STATUS
4c5c5b22-7c0e-4dcb-b3f0-b39313414898  2020-05-15T05:14:56+00:00  -         -       -       QUEUED
INFO: Refreshing access_token
Created [https://cloudbuild.googleapis.com/v1/projects/kubernetes-release-test/builds/4c5c5b22-7c0e-4dcb-b3f0-b39313414898].
Logs are available at [https://console.cloud.google.com/cloud-build/builds/4c5c5b22-7c0e-4dcb-b3f0-b39313414898?project=648026197307].
INFO: Display format: "
        table(
          id,
          createTime.date('%Y-%m-%dT%H:%M:%S%Oz', undefined='-'),
          duration(start=startTime,end=finishTime,precision=0,calendar=false,undefined="  -").slice(2:).join(""):label=DURATION,
          build_source(undefined="-"):label=SOURCE,
          build_images(undefined="-"):label=IMAGES,
          status
        )
      "
$ TOOL_ORG=justaugustus TOOL_BRANCH=rc-branch-cut krel gcbmgr --stage --type beta
INFO Verifying repository state                   
INFO Repository is in clean state                 
INFO Found matching branch "rc-branch-cut"        
INFO Found matching organization "justaugustus" and repository "release" in remote: a (git@github.com:justaugustus/release.git) 
INFO Verifying remote HEAD commit                 
INFO Got remote commit: 1b3ef841bbb0ead1a10bdd53833362a8d2c0f807 
INFO Verifying that remote commit is equal to the local one 
INFO Repository is up-to-date                     
INFO Running gcbmgr with the following options: &{Stage:true Release:false Stream:false Branch:master ReleaseType:beta BuildVersion: GcpUser: LastJobs:5 Repo:0xc000403930 Version:0xc000403940} 
INFO Build options: {  cloudbuild.yaml   kubernetes-release-test false false false   } 
INFO Retrieving Kubernetes release version for ci/latest on branch master 
INFO Retrieving Kubernetes build version from https://dl.k8s.io/ci/latest.txt... 
INFO Retrieved Kubernetes version: v1.19.0-alpha.3.261+96e13de777a9eb 
INFO Trying to get the kube-cross version for master... 
INFO Found the following kube-cross version: v1.13.9-5 
INFO Listing GCB substitutions prior to build submission... 
INFO NOMOCK_TAG:                                  
INFO TOOL_ORG: justaugustus                       
INFO TOOL_REPO: release                           
INFO TYPE_TAG: beta                               
INFO BUILD_AT_HEAD:                               
INFO BUILD_POINT: v1.19.0-alpha.3.261-96e13de777a9eb 
INFO BUILDVERSION: --buildversion=v1.19.0-alpha.3.261+96e13de777a9eb 
INFO KUBE_CROSS_VERSION: v1.13.9-5                
INFO NOMOCK:                                      
INFO TOOL_BRANCH: rc-branch-cut                   
INFO GCP_USER_TAG: stephen-k8s-at-agst-us         
INFO TYPE: beta                                   
INFO RELEASE_BRANCH: master                       
INFO Build directory: /home/augustus/go/src/k8s.io/release/gcb/stage 
INFO Config directory: /home/augustus/go/src/k8s.io/release/gcb/stage 
INFO cd-ing to build directory: /home/augustus/go/src/k8s.io/release/gcb/stage 
ID                                    CREATE_TIME                DURATION  SOURCE  IMAGES  STATUS
30b40f4b-6c7f-469d-a99f-ef4e08571f11  2020-05-15T05:15:33+00:00  -         -       -       QUEUED
Created [https://cloudbuild.googleapis.com/v1/projects/kubernetes-release-test/builds/30b40f4b-6c7f-469d-a99f-ef4e08571f11].
Logs are available at [https://console.cloud.google.com/cloud-build/builds/30b40f4b-6c7f-469d-a99f-ef4e08571f11?project=648026197307].
INFO: Display format: "
        table(
          id,
          createTime.date('%Y-%m-%dT%H:%M:%S%Oz', undefined='-'),
          duration(start=startTime,end=finishTime,precision=0,calendar=false,undefined="  -").slice(2:).join(""):label=DURATION,
          build_source(undefined="-"):label=SOURCE,
          build_images(undefined="-"):label=IMAGES,
          status
        )
      "
$ TOOL_ORG=justaugustus TOOL_BRANCH=rc-branch-cut krel gcbmgr --stage --type rc --branch release-1.19
INFO Verifying repository state                   
INFO Repository is in clean state                 
INFO Found matching branch "rc-branch-cut"        
INFO Found matching organization "justaugustus" and repository "release" in remote: a (git@github.com:justaugustus/release.git) 
INFO Verifying remote HEAD commit                 
INFO Got remote commit: 1b3ef841bbb0ead1a10bdd53833362a8d2c0f807 
INFO Verifying that remote commit is equal to the local one 
INFO Repository is up-to-date                     
INFO Running gcbmgr with the following options: &{Stage:true Release:false Stream:false Branch:release-1.19 ReleaseType:rc BuildVersion: GcpUser: LastJobs:5 Repo:0xc000403930 Version:0xc000403940} 
INFO Build options: {  cloudbuild.yaml   kubernetes-release-test false false false   } 
INFO Retrieving Kubernetes release version for ci/latest on branch release-1.19 
INFO Retrieving Kubernetes build version from https://dl.k8s.io/ci/latest-1.19.txt... 
INFO Retrieved Kubernetes version: v1.19.0-alpha.3.261+96e13de777a9eb 
INFO Trying to get the kube-cross version for release-1.19... 
INFO Error retrieving the kube-cross version for the 'release-1.19': HTTP status not OK (404) for https://raw.githubusercontent.com/kubernetes/kubernetes/release-1.19/build/build-image/cross/VERSION 
INFO Trying to get the kube-cross version for master... 
INFO Found the following kube-cross version: v1.13.9-5 
INFO Listing GCB substitutions prior to build submission... 
INFO BUILD_AT_HEAD:                               
INFO BUILDVERSION: --buildversion=v1.19.0-alpha.3.261+96e13de777a9eb 
INFO NOMOCK:                                      
INFO TOOL_REPO: release                           
INFO TYPE: rc                                     
INFO TYPE_TAG: rc                                 
INFO RELEASE_BRANCH: release-1.19                 
INFO BUILD_POINT: v1.19.0-alpha.3.261-96e13de777a9eb 
INFO KUBE_CROSS_VERSION: v1.13.9-5                
INFO NOMOCK_TAG:                                  
INFO TOOL_ORG: justaugustus                       
INFO TOOL_BRANCH: rc-branch-cut                   
INFO GCP_USER_TAG: stephen-k8s-at-agst-us         
INFO Build directory: /home/augustus/go/src/k8s.io/release/gcb/stage 
INFO Config directory: /home/augustus/go/src/k8s.io/release/gcb/stage 
INFO cd-ing to build directory: /home/augustus/go/src/k8s.io/release/gcb/stage 
ID                                    CREATE_TIME                DURATION  SOURCE  IMAGES  STATUS
c5c75071-45cc-400d-9a7d-2eff91d4b5a1  2020-05-15T05:15:48+00:00  -         -       -       QUEUED
Created [https://cloudbuild.googleapis.com/v1/projects/kubernetes-release-test/builds/c5c75071-45cc-400d-9a7d-2eff91d4b5a1].
Logs are available at [https://console.cloud.google.com/cloud-build/builds/c5c75071-45cc-400d-9a7d-2eff91d4b5a1?project=648026197307].
INFO: Display format: "
        table(
          id,
          createTime.date('%Y-%m-%dT%H:%M:%S%Oz', undefined='-'),
          duration(start=startTime,end=finishTime,precision=0,calendar=false,undefined="  -").slice(2:).join(""):label=DURATION,
          build_source(undefined="-"):label=SOURCE,
          build_images(undefined="-"):label=IMAGES,
          status
        )
      "
$ TOOL_ORG=justaugustus TOOL_BRANCH=rc-branch-cut krel gcbmgr --stage --type official --branch release-1.18
INFO Verifying repository state                   
INFO Repository is in clean state                 
INFO Found matching branch "rc-branch-cut"        
INFO Found matching organization "justaugustus" and repository "release" in remote: a (git@github.com:justaugustus/release.git) 
INFO Verifying remote HEAD commit                 
INFO Got remote commit: 1b3ef841bbb0ead1a10bdd53833362a8d2c0f807 
INFO Verifying that remote commit is equal to the local one 
INFO Repository is up-to-date                     
INFO Running gcbmgr with the following options: &{Stage:true Release:false Stream:false Branch:release-1.18 ReleaseType:official BuildVersion: GcpUser: LastJobs:5 Repo:0xc000403930 Version:0xc000403940} 
INFO Build options: {  cloudbuild.yaml   kubernetes-release-test false false false   } 
INFO Retrieving Kubernetes release version for ci/latest on branch release-1.18 
INFO Retrieving Kubernetes build version from https://dl.k8s.io/ci/latest-1.18.txt... 
INFO Retrieved Kubernetes version: v1.18.3-beta.0.43+209a8b052eadd3 
INFO Trying to get the kube-cross version for release-1.18... 
INFO Found the following kube-cross version: v1.13.9-5 
INFO Listing GCB substitutions prior to build submission... 
INFO BUILD_POINT: v1.18.3-beta.0.43-209a8b052eadd3 
INFO BUILDVERSION: --buildversion=v1.18.3-beta.0.43+209a8b052eadd3 
INFO KUBE_CROSS_VERSION: v1.13.9-5                
INFO NOMOCK_TAG:                                  
INFO TOOL_ORG: justaugustus                       
INFO TOOL_REPO: release                           
INFO GCP_USER_TAG: stephen-k8s-at-agst-us         
INFO BUILD_AT_HEAD:                               
INFO NOMOCK:                                      
INFO TOOL_BRANCH: rc-branch-cut                   
INFO TYPE: official                               
INFO TYPE_TAG: official                           
INFO RELEASE_BRANCH: release-1.18                 
INFO Build directory: /home/augustus/go/src/k8s.io/release/gcb/stage 
INFO Config directory: /home/augustus/go/src/k8s.io/release/gcb/stage 
INFO cd-ing to build directory: /home/augustus/go/src/k8s.io/release/gcb/stage 
ID                                    CREATE_TIME                DURATION  SOURCE  IMAGES  STATUS
a8a9eb48-5aa6-4de8-962a-2beffd4874f9  2020-05-15T05:16:02+00:00  -         -       -       QUEUED
Created [https://cloudbuild.googleapis.com/v1/projects/kubernetes-release-test/builds/a8a9eb48-5aa6-4de8-962a-2beffd4874f9].
Logs are available at [https://console.cloud.google.com/cloud-build/builds/a8a9eb48-5aa6-4de8-962a-2beffd4874f9?project=648026197307].
INFO: Display format: "
        table(
          id,
          createTime.date('%Y-%m-%dT%H:%M:%S%Oz', undefined='-'),
          duration(start=startTime,end=finishTime,precision=0,calendar=false,undefined="  -").slice(2:).join(""):label=DURATION,
          build_source(undefined="-"):label=SOURCE,
          build_images(undefined="-"):label=IMAGES,
          status
        )
      "
```

**Comment by justaugustus** [comment]
### [Alpha release against `master`](https://console.cloud.google.com/cloud-build/builds/4c5c5b22-7c0e-4dcb-b3f0-b39313414898?project=648026197307):

```shell
$ TOOL_ORG=justaugustus TOOL_BRANCH=rc-branch-cut krel gcbmgr --stage 
```

```shell
Step #2: ================================================================================
Step #2: SET BUILD CANDIDATE  (2/6)
Step #2: ================================================================================
Step #2: 
Step #2: RELEASE_BRANCH==master already exists
Step #2: PARENT_BRANCH set to
Step #2: BRANCH_POINT set to
Step #2: testing_branch set to master
Step #2: ATTENTION: Using --buildversion=v1.19.0-alpha.3.261+96e13de777a9eb
Step #2: JENKINS_BUILD_VERSION set to v1.19.0-alpha.3.261+96e13de777a9eb
Step #2: [2020-May-15 05:17:16 UTC] get_build_candidate in 1s
Step #2: set_release_version args: v1.19.0-alpha.3.261+96e13de777a9eb master
Step #2: RELEASE_VERSION[alpha]=v1.19.0-alpha.4
Step #2: RELEASE_VERSION_PRIME=v1.19.0-alpha.4
Step #2: ORDERED_RELEASE_KEYS[0]=alpha
Step #2: GOPATH: /workspace/anago-v1.19.0-alpha.3.261+96e13de777a9eb
Step #2: 
Step #2: ================================================================================
Step #2: SESSION VALUES
Step #2: ================================================================================
Step #2: 
Step #2: WORKDIR:                        /workspace/anago-v1.19.0-alpha.3.261+96e13de777a9eb
Step #2: TREE_ROOT:                      $WORKDIR/src/k8s.io/kubernetes
Step #2: JENKINS_BUILD_VERSION:          v1.19.0-alpha.3.261+96e13de777a9eb
Step #2: RELEASE_VERSION_PRIME:          v1.19.0-alpha.4
Step #2: RELEASE_VERSION[alpha]:         v1.19.0-alpha.4
Step #2: RELEASE_BRANCH:                 master
Step #2: GCRIO_PATH:                     gcr.io/kubernetes-release-test
Step #2: RELEASE_BUCKET:                 kubernetes-release-gcb
Step #2: BUCKET_TYPE:                    stage
Step #2: CHANGELOG_FILEPATH:             CHANGELOG/CHANGELOG-1.19.md
Step #2: FLAGS_nomock:                   0
Step #2: FLAGS_type:                     alpha
Step #2: LOGFILE:                        /workspace/tmp/anago-stage.log
Step #2: 
Step #2: ATTENTION: This is a mock (--mock) run. Publishing will be based on the above values
Step #2: for RELEASE_BUCKET, BUCKET_TYPE and GCRIO_PATH.
Step #2: 
Step #2: 
Step #2: >>>>>>>> View detailed session output with:  tailf /workspace/tmp/anago-stage.log
Step #2: >>>>>>>> (Previous logs can be found in /workspace/tmp/anago-stage.log.{1..10})
```

### [Beta release against `master`](https://console.cloud.google.com/cloud-build/builds/30b40f4b-6c7f-469d-a99f-ef4e08571f11?project=648026197307):

```shell
$ TOOL_ORG=justaugustus TOOL_BRANCH=rc-branch-cut krel gcbmgr --stage --type beta
```

```shell
Step #2: ================================================================================
Step #2: SET BUILD CANDIDATE  (2/6)
Step #2: ================================================================================
Step #2: 
Step #2: RELEASE_BRANCH==master already exists
Step #2: PARENT_BRANCH set to
Step #2: BRANCH_POINT set to
Step #2: testing_branch set to master
Step #2: ATTENTION: Using --buildversion=v1.19.0-alpha.3.261+96e13de777a9eb
Step #2: JENKINS_BUILD_VERSION set to v1.19.0-alpha.3.261+96e13de777a9eb
Step #2: [2020-May-15 05:17:48 UTC] get_build_candidate in 1s
Step #2: set_release_version args: v1.19.0-alpha.3.261+96e13de777a9eb master
Step #2: RELEASE_VERSION[beta]=v1.19.0-alpha.4
Step #2: RELEASE_VERSION_PRIME=v1.19.0-alpha.4
Step #2: ORDERED_RELEASE_KEYS[0]=beta
Step #2: GOPATH: /workspace/anago-v1.19.0-alpha.3.261+96e13de777a9eb
Step #2: 
Step #2: ================================================================================
Step #2: SESSION VALUES
Step #2: ================================================================================
Step #2: 
Step #2: WORKDIR:                        /workspace/anago-v1.19.0-alpha.3.261+96e13de777a9eb
Step #2: TREE_ROOT:                      $WORKDIR/src/k8s.io/kubernetes
Step #2: JENKINS_BUILD_VERSION:          v1.19.0-alpha.3.261+96e13de777a9eb
Step #2: RELEASE_VERSION_PRIME:          v1.19.0-alpha.4
Step #2: RELEASE_VERSION[beta]:          v1.19.0-alpha.4
Step #2: RELEASE_BRANCH:                 master
Step #2: GCRIO_PATH:                     gcr.io/kubernetes-release-test
Step #2: RELEASE_BUCKET:                 kubernetes-release-gcb
Step #2: BUCKET_TYPE:                    stage
Step #2: CHANGELOG_FILEPATH:             CHANGELOG/CHANGELOG-1.19.md
Step #2: FLAGS_nomock:                   0
Step #2: FLAGS_type:                     beta
Step #2: LOGFILE:                        /workspace/tmp/anago-stage.log
Step #2: 
Step #2: ATTENTION: This is a mock (--mock) run. Publishing will be based on the above values
Step #2: for RELEASE_BUCKET, BUCKET_TYPE and GCRIO_PATH.
Step #2: 
Step #2: 
Step #2: >>>>>>>> View detailed session output with:  tailf /workspace/tmp/anago-stage.log
Step #2: >>>>>>>> (Previous logs can be found in /workspace/tmp/anago-stage.log.{1..10})
```

### [RC against `release-1.19`](https://console.cloud.google.com/cloud-build/builds/c5c75071-45cc-400d-9a7d-2eff91d4b5a1?project=648026197307):

```shell
$ TOOL_ORG=justaugustus TOOL_BRANCH=rc-branch-cut krel gcbmgr --stage --type rc --branch release-1.19
```

```shell
Step #2: ================================================================================
Step #2: SET BUILD CANDIDATE  (2/6)
Step #2: ================================================================================
Step #2: 
Step #2: RELEASE_BRANCH==release-1.19 does not yet exist
Step #2: PARENT_BRANCH set to master
Step #2: BRANCH_POINT set to
Step #2: testing_branch set to master
Step #2: ATTENTION: Using --buildversion=v1.19.0-alpha.3.261+96e13de777a9eb
Step #2: JENKINS_BUILD_VERSION set to v1.19.0-alpha.3.261+96e13de777a9eb
Step #2: [2020-May-15 05:18:21 UTC] get_build_candidate in 1s
Step #2: set_release_version args: v1.19.0-alpha.3.261+96e13de777a9eb release-1.19 master
Step #2: RELEASE_VERSION[alpha]=v1.20.0-alpha.0
Step #2: RELEASE_VERSION[rc]=v1.19.0-rc.0
Step #2: RELEASE_VERSION_PRIME=v1.19.0-rc.0
Step #2: ORDERED_RELEASE_KEYS[0]=rc
Step #2: ORDERED_RELEASE_KEYS[1]=alpha
Step #2: GOPATH: /workspace/anago-v1.19.0-alpha.3.261+96e13de777a9eb
Step #2: 
Step #2: ================================================================================
Step #2: SESSION VALUES
Step #2: ================================================================================
Step #2: 
Step #2: WORKDIR:                        /workspace/anago-v1.19.0-alpha.3.261+96e13de777a9eb
Step #2: TREE_ROOT:                      $WORKDIR/src/k8s.io/kubernetes
Step #2: PARENT_BRANCH:                  master
Step #2: JENKINS_BUILD_VERSION:          v1.19.0-alpha.3.261+96e13de777a9eb
Step #2: RELEASE_VERSION_PRIME:          v1.19.0-rc.0
Step #2: RELEASE_VERSION[alpha]:         v1.20.0-alpha.0
Step #2: RELEASE_VERSION[rc]:            v1.19.0-rc.0
Step #2: RELEASE_BRANCH:                 release-1.19
Step #2: GCRIO_PATH:                     gcr.io/kubernetes-release-test
Step #2: RELEASE_BUCKET:                 kubernetes-release-gcb
Step #2: BUCKET_TYPE:                    stage
Step #2: CHANGELOG_FILEPATH:             CHANGELOG/CHANGELOG-1.19.md
Step #2: FLAGS_nomock:                   0
Step #2: FLAGS_type:                     rc
Step #2: LOGFILE:                        /workspace/tmp/anago-stage.log
Step #2: 
Step #2: ATTENTION: release-1.19 is a NEW branch off master!
Step #2: 
Step #2: ATTENTION: This is a mock (--mock) run. Publishing will be based on the above values
Step #2: for RELEASE_BUCKET, BUCKET_TYPE and GCRIO_PATH.
Step #2: 
Step #2: 
Step #2: >>>>>>>> View detailed session output with:  tailf /workspace/tmp/anago-stage.log
Step #2: >>>>>>>> (Previous logs can be found in /workspace/tmp/anago-stage.log.{1..10})
```

### [Official release against `release-1.18`](https://console.cloud.google.com/cloud-build/builds/a8a9eb48-5aa6-4de8-962a-2beffd4874f9?project=648026197307):

```shell
$ TOOL_ORG=justaugustus TOOL_BRANCH=rc-branch-cut krel gcbmgr --stage --type official --branch release-1.18
```

```shell
Step #2: ================================================================================
Step #2: SET BUILD CANDIDATE  (2/6)
Step #2: ================================================================================
Step #2: 
Step #2: RELEASE_BRANCH==release-1.18 already exists
Step #2: PARENT_BRANCH set to
Step #2: BRANCH_POINT set to
Step #2: testing_branch set to release-1.18
Step #2: ATTENTION: Using --buildversion=v1.18.3-beta.0.43+209a8b052eadd3
Step #2: JENKINS_BUILD_VERSION set to v1.18.3-beta.0.43+209a8b052eadd3
Step #2: [2020-May-15 05:18:27 UTC] get_build_candidate in 1s
Step #2: set_release_version args: v1.18.3-beta.0.43+209a8b052eadd3 release-1.18
Step #2: RELEASE_VERSION[official]=v1.18.3
Step #2: RELEASE_VERSION[rc]=v1.18.4-rc.0
Step #2: RELEASE_VERSION_PRIME=v1.18.3
Step #2: ORDERED_RELEASE_KEYS[0]=official
Step #2: ORDERED_RELEASE_KEYS[1]=rc
Step #2: GOPATH: /workspace/anago-v1.18.3-beta.0.43+209a8b052eadd3
Step #2: 
Step #2: ================================================================================
Step #2: PENDING PRS ON THE release-1.18 BRANCH
Step #2: ================================================================================
Step #2: 
Step #2: <snip>
Step #2: 
Step #2: ================================================================================
Step #2: SESSION VALUES
Step #2: ================================================================================
Step #2: 
Step #2: WORKDIR:                        /workspace/anago-v1.18.3-beta.0.43+209a8b052eadd3
Step #2: TREE_ROOT:                      $WORKDIR/src/k8s.io/kubernetes
Step #2: JENKINS_BUILD_VERSION:          v1.18.3-beta.0.43+209a8b052eadd3
Step #2: RELEASE_VERSION_PRIME:          v1.18.3
Step #2: RELEASE_VERSION[official]:      v1.18.3
Step #2: RELEASE_VERSION[rc]:            v1.18.4-rc.0
Step #2: RELEASE_BRANCH:                 release-1.18
Step #2: GCRIO_PATH:                     gcr.io/kubernetes-release-test
Step #2: RELEASE_BUCKET:                 kubernetes-release-gcb
Step #2: BUCKET_TYPE:                    stage
Step #2: CHANGELOG_FILEPATH:             CHANGELOG/CHANGELOG-1.18.md
Step #2: FLAGS_nomock:                   0
Step #2: FLAGS_type:                     official
Step #2: LOGFILE:                        /workspace/tmp/anago-stage.log
Step #2: 
Step #2: ATTENTION: This is a mock (--mock) run. Publishing will be based on the above values
Step #2: for RELEASE_BUCKET, BUCKET_TYPE and GCRIO_PATH.
Step #2: 
Step #2: 
Step #2: >>>>>>>> View detailed session output with:  tailf /workspace/tmp/anago-stage.log
Step #2: >>>>>>>> (Previous logs can be found in /workspace/tmp/anago-stage.log.{1..10})
```

`alpha`: Session values look good
`beta`: Release label is right, version is wrong
`rc`: Session values look good
`official`: Session values look good

A few things we need to do next:

- [ ] Fix version logic on `beta` cuts
- [ ] Ensure alphas of the same x.y can't be cut after a beta has been cut

File: lib/releaselib.sh [hunk 1]
```diff
@@ -449,15 +435,9 @@ release::set_release_version () {
...
-      if [[ "${build_version[label]}" == "-rc" ]]; then
-        RELEASE_VERSION[rc]+="-rc.$((${build_version[labelid]}+1))"
-      else
-        # Start release candidates at 1 instead of 0
-        RELEASE_VERSION[rc]+="-rc.1"
-      fi
+      RELEASE_VERSION[rc]+="-rc.$((${build_version[labelid]}+1))"
       RELEASE_VERSION_PRIME="${RELEASE_VERSION[rc]}"
     fi
   elif [[ "$FLAGS_type" == "beta" ]]; then
```
Comment (saschagrunert) [inline lib/releaselib.sh:443]: Theoretically, if this code path gets executed during the beta cut, then we should get the correct version, right?

**Comment by justaugustus** [comment]
https://console.cloud.google.com/cloud-build/builds/85810408-5263-4197-b990-0e9a578543c8?project=kubernetes-release-test shows a successful beta stage against master:

```shell
$ TOOL_ORG=justaugustus TOOL_BRANCH=rc-branch-cut krel gcbmgr --stage --type beta
```

```shell
Step #2: ================================================================================
Step #2: SET BUILD CANDIDATE  (2/6)
Step #2: ================================================================================
Step #2: 
Step #2: RELEASE_BRANCH==master already exists
Step #2: PARENT_BRANCH set to
Step #2: BRANCH_POINT set to
Step #2: testing_branch set to master
Step #2: ATTENTION: Using --buildversion=v1.19.0-alpha.3.261+96e13de777a9eb
Step #2: JENKINS_BUILD_VERSION set to v1.19.0-alpha.3.261+96e13de777a9eb
Step #2: [2020-May-15 07:17:16 UTC] get_build_candidate in 1s
Step #2: set_release_version args: v1.19.0-alpha.3.261+96e13de777a9eb master
Step #2: RELEASE_VERSION[beta]=v1.19.0-beta.0
Step #2: RELEASE_VERSION_PRIME=v1.19.0-beta.0
Step #2: ORDERED_RELEASE_KEYS[0]=beta
Step #2: GOPATH: /workspace/anago-v1.19.0-alpha.3.261+96e13de777a9eb
Step #2: 
Step #2: ================================================================================
Step #2: SESSION VALUES
Step #2: ================================================================================
Step #2: 
Step #2: WORKDIR:                        /workspace/anago-v1.19.0-alpha.3.261+96e13de777a9eb
Step #2: TREE_ROOT:                      $WORKDIR/src/k8s.io/kubernetes
Step #2: JENKINS_BUILD_VERSION:          v1.19.0-alpha.3.261+96e13de777a9eb
Step #2: RELEASE_VERSION_PRIME:          v1.19.0-beta.0
Step #2: RELEASE_VERSION[beta]:          v1.19.0-beta.0
Step #2: RELEASE_BRANCH:                 master
Step #2: GCRIO_PATH:                     gcr.io/kubernetes-release-test
Step #2: RELEASE_BUCKET:                 kubernetes-release-gcb
Step #2: BUCKET_TYPE:                    stage
Step #2: CHANGELOG_FILEPATH:             CHANGELOG/CHANGELOG-1.19.md
Step #2: FLAGS_nomock:                   0
Step #2: FLAGS_type:                     beta
Step #2: LOGFILE:                        /workspace/tmp/anago-stage.log
Step #2: 
Step #2: ATTENTION: This is a mock (--mock) run. Publishing will be based on the above values
Step #2: for RELEASE_BUCKET, BUCKET_TYPE and GCRIO_PATH.
Step #2: 
Step #2: 
Step #2: >>>>>>>> View detailed session output with:  tailf /workspace/tmp/anago-stage.log
Step #2: >>>>>>>> (Previous logs can be found in /workspace/tmp/anago-stage.log.{1..10})
```

**Pull request merged**